app/helpers/form_helper.rb

```diff
@@ -202,17 +202,27 @@ def file_field_f(f, attr, options = {})
   end
 
   def autocomplete_f(f, attr, options = {})
```

Comment:
This approach is convenient. no doubt. however I think in the long term, using rails helpers for react would prevent others from using react

Reply:
Interesting opinion,
@ohadlevy @tbrisker what are your thoughts ?

Reply:
I don't think there is any value in forcing developers to "mess their hands" in react just to implement a simple form field, just as the current implementation doesn't require them to "mess their hands" in jquery to get the autocomplete working - it's just an abstraction that makes development easy.
These helpers allow us to have easy to use elements with consistent layout across the whole app. Modifying helpers to use react implementation gives us the ability to change all the consumers of the helpers at once instead of having to rewrite every consumer in every plugin. It also prevents a situation where some fields of the same type have one layout while others have a different one because they use a different implementation (as we've seen when using angular implementations in katello).

Reply:
> Modifying helpers to use react implementation gives us the ability to change all the consumers of the helpers at once instead of having to rewrite every consumer in every plugin.

The question is: Do we want that? Maybe plugins use the helpers and have custom jQuery-JS-Code that relies on the form not being React.
I don't think we should change the implementation of existing fields but rather add new fields and deprecate the old ones properly.

Reply:
If some plugin modified or injects into internals of existing helper, I think it's sad but not something that should prevent us from changing the implementation. It would tie our hands to tight if we couldn't change. As long as it does not enforce my plugin to change anything if I'm using core helpers, I think that's fine.

Of course I may be missing some impact. Do you have some specific example?

Reply:
I don't have something specific. So let's continue and I'll complain afterward that I told you so ;-)

Reply:
So can we agree that when replacing existing implementations we'll keep the helpers name,
and when creating a new one, we will call it the react way? e.g: `mount_autocomplete`